zerver/lib/import_realm.py

```diff
@@ -952,6 +953,39 @@ def do_import_realm(import_dir: Path, subdomain: str, processes: int = 1) -> Rea
     realm.signup_notifications_stream_id = None
     realm.save()
 
+    # We don't import the UserProfile model yet, but we need UserProfile model
+    # IDs for direct_members field in UserGroup model.
+    update_model_ids(UserProfile, data, "user_profile")
+    if "zerver_usergroup" in data:
```

Comment:
I think it does make sense to import UserGroup before Stream, but I think we might want to take the approach already present for how the Realm=>Stream foreign keys are handled:

```
    # We don't import the Stream model yet, since it depends on Realm,                                   
    # which isn't imported yet.  But we need the Stream model IDs for                                    
    # notifications_stream.                                                                              
    update_model_ids(Stream, data, "stream")                                                             
    re_map_foreign_keys(data, "zerver_realm", "notifications_stream", related_table="stream")            
    re_map_foreign_keys(data, "zerver_realm", "signup_notifications_stream", related_table="stream")     

    ...
   
     if realm.notifications_stream_id is not None:                                                        
        notifications_stream_id: Optional[int] = int(realm.notifications_stream_id)                      
    else:                                                                                                
        notifications_stream_id = None                                                                   
    realm.notifications_stream_id = None                                                                 
    if realm.signup_notifications_stream_id is not None:                                                 
        signup_notifications_stream_id: Optional[int] = int(realm.signup_notifications_stream_id)        
    else:                                                                                                
        signup_notifications_stream_id = None                                                            
    realm.signup_notifications_stream_id = None                                                          
    realm.save()                                                                                         
```

Given that our medium-term plans involve hanging additional `group_id` format fields off of Realm, Stream, and UserGroup itself, I think we may be happier with using that model, since we're going to have to use it for any future `Realm=>UserGroup` foreign keys anyway.

Reply:
I am not sure whether we can do this like how Realm=>Stream foreign keys are handled, because for now I am assuming that we will keep this field not-null and what initial values we should set while first creating streams like we set `None` in the code snippet above. We were [discussing](https://chat.zulip.org/#narrow/stream/3-backend/topic/default.20value.20for.20a.20setting.20based.20on.20user.20groups/near/1398310) whether we should allow null for this setting for the case like nobody has permission to do that task, but we didn't come to conclusion then.

Note - the current code in this PR is not correct. Though it works and pass tests for now, it is technically incorrect and the tests pass because we bulk import the stream model using `bulk_create` which does not call `save`.

Reply:
I think for the purpose of the import code, we can set the field to `None` if it's unset, and then have an assert later in the import tool to confirm it got filled in. 

And then the second thing we'll need is a default for what to fill in for this setting later in the import process -- that's probably the "admins" system group, to match the existing permissions?

Reply:
Posted https://chat.zulip.org/#narrow/stream/378-api-design/topic/default.20value.20for.20a.20setting.20based.20on.20user.20groups/near/1432671 as a proposal; I think for this logic, that means we can temporarily set values to `None`, but assert that they are filled in before the end of the import process.